src/cpp/session/modules/SessionEnvironment.R

```diff
@@ -465,67 +471,75 @@
 .rs.addFunction("valueDescription", function(obj)
 {
    tryCatch(
+      .rs.valueDescriptionImpl(obj),
+      error = function(e) ""
+   )
+})
+
+.rs.addFunction("valueDescriptionImpl", function(obj)
+{
+   if (missing(obj))
    {
-      if (missing(obj))
-      {
-        return("Missing argument")
-      }
-      else if (is(obj, "ore.frame"))
-      {
-        sqlTable <- attr(obj, "sqlTable", exact = TRUE)
-        if (is.null(sqlTable))
-          return("Oracle R frame") 
-        else
-          return(paste("Oracle R frame:", sqlTable))
-      }
-      else if (.rs.isExternalPointer(obj))
-      {
-         class <- class(obj)
-         if (length(class) && !identical(class, "externalptr"))
-         {
-            fmt <- "External pointer of class '%s'"
-            return(sprintf(fmt, class[[1]]))
-         }
-         else
-         {
-            return("External pointer")
-         }
-      }
-      else if (is.data.frame(obj))
-      {
-         return(paste(dim(obj)[1],
-                      "obs. of",
-                      dim(obj)[2],
-                      ifelse(dim(obj)[2] == 1, "variable", "variables"),
-                      sep=" "))
-      }
-      else if (is.environment(obj))
-      {
-         return("Environment")
-      }
-      else if (isS4(obj))
-      {
-         return(paste("Formal class ", is(obj)))
-      }
-      else if (is.list(obj))
+      return("Missing argument")
+   }
+   else if (is.null(obj))
```

Comment:
Note: this is the pertinent change in this branch for the PR; the rest is just re-formatted from the old code.